### DISCUSSION TITLE
Failed to install ms-toolsai.jupyter plugin in the browser-only environment

### DISCUSSION THREAD

**Discussion by dian533** [eclipse-theia/theia#13938] [Q&A]
I have successfully installed the vscode.ipynb plugin in the browser-only Theia environment using the [browser-only-plugins-poc](https://github.com/eclipsesource/theia/tree/browser-only-plugins-poc) example. However, when I try to install the ms-toolsai.jupyter plugin using a similar approach, I encounter the following error. 

![image](https://github.com/user-attachments/assets/f9e13130-cb6c-47cf-997a-f401027a6cc9)

The file has been downloaded successfully
![image](https://github.com/user-attachments/assets/4da1cd5a-5f5c-4873-8a7c-be7b24bef998)

How can I resolve this error and successfully install the extension?

**Comment by msujew** [comment]
Hey @dian533,

I can reproduce the issue, but it's very likely that Theia is not the culprit here. The browser bundle of the jupyter extension contains references to the `module` object, which is only available in CommonJS/Node.js. The browser attempts to load the JavaScript in the file but fails due to object accesses on `undefined`. This in turn leads to the error you're seeing. I believe that the Jupyter extension versions available on open-vsx are not correctly compiled.

Note that the Jupyter extension on open-vsx is provided by a third party and not by Microsoft themselves, see also https://github.com/open-vsx/publish-extensions/issues/740 and https://github.com/open-vsx/publish-extensions/issues/703. Since there are a few build issues outlined in the linked issue for the extension, it's possible that the browser build also isn't correctly compiled.

Running the officially distributed web extension available from [the vscode marketplace](https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-toolsai/vsextensions/jupyter/2024.6.2024070901/vspackage?targetPlatform=web) works as expected. So it's not a Theia issue.

**Reply by dian533** [reply]
Okay, thanks for your suggestion.
